### PR TITLE
Update Workflows for RPC Only Release

### PR DESCRIPTION
**Pull request by stellarsaur**
### What

Related to the [RPC/CLI repo split](https://github.com/stellar/soroban-tools/issues/1171).

This PR prepares GitHub workflows to release Soroban RPC from the `soroban-rpc` repo, rather than packaged with the CLI in the `soroban-tools` repo.

### Why

See [this issue](https://github.com/stellar/soroban-tools/issues/1115) for motivation behind splitting up the repos and release processes.

### CODE REVIEW

**Comment by 2opremio** [comment]
Nice docs!!!

**Review by 2opremio — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -16,9 +16,10 @@ jobs:
       # the gh tag of system-test repo version to run
       SYSTEM_TEST_GIT_REF: master
 
-      # the soroban tools source code to compile and run from system test
+      # the soroban CLI & RPC source code to compile and run from system test
       # refers to checked out source of current git hub ref context
-      SYSTEM_TEST_SOROBAN_TOOLS_REF: ${{ github.workspace }}/soroban-tools
+      SYSTEM_TEST_SOROBAN_CLI_REF: https://github.com/stellar/soroban-tools.git\#main
+      SYSTEM_TEST_SOROBAN_RPC_REF: https://github.com/stellar/soroban-rpc.git\#main
```
Comment (sreuland) [inline .github/workflows/e2e.yml:22]: you want to build the rpc code from this repo correct? this path should point to gha workspace which has the branch code checked out locally:
```
${{ github.workspace }}/soroban-rpc
```

File: RELEASING.md [hunk 2]
```diff
@@ -1,5 +1,58 @@
...
+    * Set `VERSION_OVERRIDE` if necessary, replacing "-" with "~" (e.g. 20.0.0~rc1)
+    * Set `BRANCH` to the release tag (e.g. v20.0.0-rc1)
+    * Enable `PROMOTE_TO_TESTING`
+    * ![Pipeline Parameters](docs/img/soroban-package-builder.png)
+
+2. Review the package stage log, and note the build number for next steps
+
+![Soroban Package Logs](docs/img/soroban-package-logs.png)
+
+## Push New Docker Image
```
Comment (mollykarcher) [inline RELEASING.md:35]: There is also a promote-to-stable jenkins job that needs to be run after the apt and docker image are generated. Probably want a new section for that

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (stellarsaur) [reply .github/workflows/e2e.yml]: Yes, I would like to build from this repo. Strangely enough, `${{ github.workspace }}/soroban-rpc` produce the same error discussed in [this thread](https://stellarfoundation.slack.com/archives/G01FUHTUQ8Z/p1706656493328379). Looking into this more

Comment (stellarsaur) [reply .github/workflows/e2e.yml]: The issue was not adding a checkout action for `soroban-rpc`

**Pull request merged**